File: src/Model/ISeaGrid.vb
Public Interface ISeaGrid

    ReadOnly Property Width() As Integer

    ReadOnly Property Height() As Integer

    Event Changed As EventHandler

    ReadOnly Property Item(ByVal row As Integer, ByVal column As Integer) As TileView

    Function HitTile(ByVal row As Integer, ByVal col As Integer) As AttackResult
End Interface

File: src/Model/ISeaGrid.cs
using System;
using System.Collections;
using System.Collections.Generic;
public interface ISeaGrid {

	int Width { get; }

	int Height { get; }

	event EventHandler Changed;
	
	TileView this[int row, int column] { get; }
	
	AttackResult HitTile(int row, int col);
}